internal/gatewayapi/tls.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// validateTLSSecretData ensures the cert and key provided in a secret
+// is not malformed and can be properly parsed.
+func validateTLSSecretData(secret *corev1.Secret) error {
+
+	certData := secret.Data[corev1.TLSCertKey]
+
+	certBlock, _ := pem.Decode(certData)
+	if certBlock == nil {
+		return fmt.Errorf("unable to parse pem data in %s", corev1.TLSCertKey)
+	}
+
+	_, err := x509.ParseCertificate(certBlock.Bytes)
+	if err != nil {
+		return fmt.Errorf("unable to parse certificate in %s: %w", corev1.TLSCertKey, err)
+	}
+
+	keyData := secret.Data[corev1.TLSPrivateKeyKey]
+
+	keyBlock, _ := pem.Decode(keyData)
+	if keyBlock == nil {
+		return fmt.Errorf("unable to decode pem data in %s", corev1.TLSCertKey)
```

Comment:
same as above, lets not log the secret content, but maybe the name of the secret and unparsable attribute

Reply:
🙈